lib/puppet/util/windows/process.rb

```diff
@@ -97,6 +124,30 @@ def with_process_token(access, &block)
   end
   module_function :with_process_token
 
+  def get_process_image_name_by_pid(pid)
+    image_name = ""
+
+    open_process(PROCESS_QUERY_INFORMATION, FFI::WIN32_FALSE, pid) do |phandle|
+
+      FFI::MemoryPointer.new(:pdword, 1) do |exe_name_length_ptr|
+        exe_name_length_ptr.write_dword(MAX_PATH_LENGTH)
+        FFI::MemoryPointer.new(:lpwstr, exe_name_length_ptr.read_dword) do |exe_name_ptr|
```

Comment:
I think it's probably simpler to just reuse the constant here for the size, rather than reading the value behind the FFI pointer.